UserContext/RoleProvider.php

```diff
@@ -34,10 +35,18 @@ class RoleProvider implements ContextProviderInterface
      * firewall. It is however not valid to call updateUserContext when not in
      * a firewall context.
      *
-     * @param SecurityContextInterface|null $context
+     * @param SecurityContextInterface|TokenStorageInterface $context
      */
-    public function __construct(SecurityContextInterface $context = null)
+    public function __construct($context = null)
     {
+        if ($context && !$context instanceof SecurityContextInterface
```

Comment:
lets put the `&& ...` on a separate line to not create the impression those two conditions have a different priority than the last one.